src/lib/mongoBackend/MongoCommonRegister.cpp

```diff
@@ -226,10 +226,15 @@ static bool addTriggeredSubscriptions
   auto_ptr<DBClientCursor> cursor;
   BSONObj                  query = BSON("$or" << BSON_ARRAY(queryNoPattern.obj() << queryPattern.obj()));
 
-  if (!collectionQuery(getSubscribeContextAvailabilityCollectionName(tenant), query, &cursor, &err))
+  TIME_STAT_MONGO_READ_WAIT_START();
+  DBClientBase* connection = getMongoConnection();
+  if (!collectionQuery(connection, getSubscribeContextAvailabilityCollectionName(tenant), query, &cursor, &err))
   {
+    TIME_STAT_MONGO_READ_WAIT_STOP();
+    releaseMongoConnection(connection);
     return false;
   }
+  TIME_STAT_MONGO_READ_WAIT_STOP();
```

Comment:
I have some doubts regarding the right moment to do the TIME_STAT_MONGO_READ_WAIT_STOP()

1. Just after the collectionQuery (as it is now)
2. After processing cursor results (i.e. it would be just after releaseMongoConnection() in line 277)

I preferred alternative 1, as alternative 2 is highly variable on the particular cursor processing, thus if we do in that way, then the mongoWait metric will be "mixed" with mongoBackend processing.